packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1551,8 +1552,9 @@ export class KeyringController extends BaseController<
   ): Promise<CallbackResult> {
     this.#assertIsUnlocked();
 
-    return this.#persistOrRollback(async () => {
+    return this.#withRollback(async () => {
```

Comment:
Thoughts on moving the check in `#persistOrRollback` instead of using `#withRollback` here? it would avoid giving to `withKeyring` the responsibility to update the vault directly

Reply:
Yes, that was my initial idea but the thing is I need to have a reference of the `keyring` to be able to serialize it.

So this would probably require to move the "keyring selector" to that method?

I also thought of adding a new `persistKeyringOrRollback` to not pollute the existing `persistOrRollback`.

Any preferences?

Reply:
We wouldn't check a single keyring in `#persistOrRollback` (as some operations need to interact with multiple keyrings) - we would have to use `this.#getSerializedKeyrings()` before and after executing the callback passed to `#persistOrRollback`

Reply:
That makes sense yes, I just went with the "fastest option" (being, just serializing + comparing what we need in that case)

But having this in `#persistOrRollback` is an option too, just potentially a bit slower, but maybe we wouldn't notice the difference.

Reply:
Resolving, since I've moved most of the conditional logic to `#persistOrRollback` now.